### DISCUSSION TITLE
I have a DataGrid inside ScrollViewer. How do I prevent DataGrid row selection changing scroll position?

### DISCUSSION THREAD

**Discussion by arnirichard** [AvaloniaUI/Avalonia#14449] [Q&A]
Questions:

1. See title
2. Is there a way to actually disable row selection in DataGrid?
3. I have a button in a DataGridTemplateColumn, how can I make button presses possible? Currently, the DataGridRow will swallow the click event

**Comment by timunie** [comment]
> I have a DataGrid inside ScrollViewer.

Better not doing this unless you specify fixed Height for DataGrid. 

> I have a button in a DataGridTemplateColumn, how can I make button presses possible? Currently, the DataGridRow will swallow the click event

Consider using a Command instead

**Reply by arnirichard** [reply]
I have a fixed height for the datagrid, so how can I prevent the DataGrid from changing the scroll position?

**Reply by timunie** [reply]
I cannot really follow you. Do you have a ScreenShot for me?

**Reply by arnirichard** [reply]
Never mind, I found out that setting Focusable to false on the DataGrid solves the problem.

**Reply by arnirichard** [reply]
I have added a command, but I need to press the button a few times before the handler is hit. It is the same problem as when I used the Tapped event.

**Reply by arnirichard** [reply]
Never mind, the reason was that the Button had transparent background.

**Reply by timunie** [reply]
transparent should be okay while null is not

**Reply by arnirichard** [reply]
yes, you are right, it was null. Good to know and thanks for your help.
